core/jvm/src/main/scala/zio/sql/expr.scala

```diff
@@ -108,7 +108,10 @@ trait ExprModule extends NewtypesModule with FeaturesModule with OpsModule {
 
     def typeTagOf[A](expr: Expr[_, _, A]): TypeTag[A] = expr.asInstanceOf[InvariantExpr[_, _, A]].typeTag
 
-    implicit def literal[A: TypeTag](a: A): Expr[Features.Literal, Any, A] = Expr.Literal(a)
+    implicit def literal[A](a: A)(implicit typeTag: TypeTag[A]): Expr[Features.Literal, Any, A] = typeTag match {
+      case TypeTag.TInstant => Expr.literal(s"TIMESTAMP '$a'").asInstanceOf[Expr[Features.Literal, Any, A]]
```

Comment:
Can you revert this change please? see comment at https://github.com/zio/zio-sql/pull/307#discussion_r528286267